flight_manager.py

```diff
@@ -125,10 +131,7 @@ def run_threads(self, sim: bool) -> None:
         except KeyboardInterrupt:
             # Ctrl-C was pressed
             logging.info("Ctrl-C Pressed, forcing drone to land")
-            comm_obj.state = StateEnum.Land
-            flight_process = self.init_flight(flight_args)
-            flight_process.start()
-
+            drone.action.return_to_launch()
```

Comment:
If you want to keep the return_to_launch functionality, you'd have to change the global flight mode instead of creating a new drone object here. This could be done in the Land state, and if so, we'd need to send to the Land state that Ctrl+C was pressed. However I'm pretty sure this can be done here by accessing the flight process that is created here.